### PR TITLE
fix: OIDC issuer URL with multi-tenant URL

### PR DESCRIPTION
**Pull request by jgribonvald**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [ ] tests are included
-   [ ] documentation is changed or added
-   [ ] [message properties][] have been updated with new phrases
-   [ ] view conforms with [WCAG 2.0 AA][]

##### Description of change

This fix the issuer on OIDCTokens that wasn't rewrited due to portal.protocol.server.context=${portal.protocol}://_CURRENT_SERVER_NAME_${portal.context} 
where a rewrite of the text variable `_CURRENT_SERVER_NAME_` was missing.
Cause to this fix a refactor was needed to avoid circular dependency, and nearly all the stuff for URL Customization was moved to a new module.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: uPortal-utils/uPortal-utils-url/build.gradle [hunk 1]
```diff
@@ -0,0 +1,13 @@
+description = "Apereo uPortal Utilities Url"
+
+dependencies {
+
+
+    compile "org.slf4j:slf4j-api:${slf4jVersion}"
+    compile "org.springframework:spring-beans:${springVersion}"
```
[the remark this answers is not in the document]
Comment (jgribonvald) [reply uPortal-utils/uPortal-utils-url/build.gradle:7]: @sonatype-lift ignore

File: uPortal-security/uPortal-security-core/src/main/java/org/apereo/portal/security/oauth/IdTokenFactory.java [hunk 2]
```diff
@@ -63,6 +66,8 @@ public class IdTokenFactory {
 
     @Autowired private IPersonAttributeDao personAttributeDao;
 
+    @Autowired private UrlAuthCustomizerRegistry urlAuthCustomizerRegistry;
```
[the remark this answers is not in the document]
Comment (jgribonvald) [reply uPortal-security/uPortal-security-core/src/main/java/org/apereo/portal/security/oauth/IdTokenFactory.java]: @sonatype-lift ignore 
Fixed !

File: uPortal-utils/uPortal-utils-url/build.gradle [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    compile "org.springframework:spring-core:${springVersion}"
+
+    compileOnly "org.apache.portals.pluto:pluto-container:${plutoVersion}"
+    compileOnly "${portletApiDependency}"
+    compileOnly "${servletApiDependency}"
+}
```
Comment (ChristianMurphy) [inline uPortal-utils/uPortal-utils-url/build.gradle:13]:
```suggestion
}

```

File: uPortal-utils/uPortal-utils-url/build.gradle [hunk 4]
```diff
@@ -0,0 +1,13 @@
+description = "Apereo uPortal Utilities Url"
+
+dependencies {
+
+
```
Comment (ChristianMurphy) [inline uPortal-utils/uPortal-utils-url/build.gradle:5]:
```suggestion
dependencies {
```

File: uPortal-webapp/build.gradle [hunk 5]
```diff
@@ -156,4 +157,4 @@ war {
         from "${buildDir}/generated-sources/skin/main/webapp"
         into '/'
     }
-}
+}
```
Comment (ChristianMurphy) [inline uPortal-webapp/build.gradle:160]:
```suggestion
}

```

File: uPortal-webapp/src/main/resources/properties/contexts/mvcContext.xml [hunk 6]
```diff
@@ -115,4 +121,4 @@
     </bean>
 
     <bean id="localeResolver" class="org.apereo.portal.i18n.LocaleManagerLocaleResolver" />
-</beans>
+</beans>
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/resources/properties/contexts/mvcContext.xml:124]:
```suggestion
</beans>

```

File: uPortal-security/uPortal-security-core/build.gradle [hunk 7]
```diff
@@ -16,4 +17,4 @@ dependencies {
 
     compileOnly "${portletApiDependency}"
     compileOnly "${servletApiDependency}"
-}
+}
```
Comment (ChristianMurphy) [inline uPortal-security/uPortal-security-core/build.gradle:20]:
```suggestion
}

```

File: settings.gradle [hunk 8]
```diff
@@ -57,5 +57,5 @@ include 'uPortal-hibernate:uPortal-hibernate4-dialects'
 
 include 'uPortal-utils:uPortal-utils-core'
 include 'uPortal-utils:uPortal-utils-jmx'
-include 'uPortal-health'
-
+include 'uPortal-utils:uPortal-utils-url'
+include 'uPortal-health'
```
Comment (ChristianMurphy) [inline settings.gradle:61]:
```suggestion
include 'uPortal-health'

```

File: uPortal-utils/uPortal-utils-url/build.gradle (hunk 3 above)
Comment (jgribonvald) [reply uPortal-utils/uPortal-utils-url/build.gradle]: Fixed

File: uPortal-webapp/src/main/resources/properties/contexts/mvcContext.xml (hunk 6 above)
Comment (jgribonvald) [reply uPortal-webapp/src/main/resources/properties/contexts/mvcContext.xml]: fixed

File: uPortal-utils/uPortal-utils-url/build.gradle (hunk 4 above)
Comment (jgribonvald) [reply uPortal-utils/uPortal-utils-url/build.gradle]: fixed

**Comment by jgribonvald** [comment]
Thanks @ChristianMurphy I forgot to check my `git add -A`, and on some conf IntellijIdea is configured to remove blank line at end of files...
So all removed blank lines are fixed

**Comment by jgribonvald** [comment]
Hi folks, can I merge this PR ? If no more feedback I will do as it fix a old thing.
Thanks

**Pull request merged**